src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -576,6 +603,15 @@ export class MockHttpSocket extends MockSocket {
     if (this.requestStream) {
       this.requestStream.push(null)
     }
+    this.headers = []
+    this.url = ''
+  }
+
+  private onResponseHeaders(headers: Array<string>, url: string): void {
+    // Once we exceeded headers limit - stop collecting them
+    if (this.responseParser.maxHeadersPair <= 0 || this.headers.length < this.responseParser.maxHeadersPair) { 
```

Comment:
I'm not adding this handling in the proposed changes because I think this isn't something we should handle. I'm happy to be proven wrong though.

Reply:
I agree. It's not really necessary